### PR TITLE
Implement Authentication

### PR DESCRIPTION
**Pull request by cquiro**
## Summary

- Add devise_token_auth gem to Gemfile.
- Change devise gem version for compatibility with devise_token_auth gem and bundle install.
- Install devise_token_auth using the generator provided with the gem.
- Edit generated migration (since I already have a User model) and run migrations.
- Add first_name and last_name to sign_up permitted params.

## Trello Card

https://trello.com/c/gGYxOQmq/25-implementar-autenticaci%C3%B3n

### CODE REVIEW

**Review by lmasello — APPROVED** [review]
Great :+1: See if we can remove the ´# frozen_string_literal: true´ from devise_token_auth.rb

File: db/migrate/20180920205430_add_devise_token_auth_info_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class AddDeviseTokenAuthInfoToUsers < ActiveRecord::Migration[5.2]
+  def change
+    change_table(:users) do |t|
```
Comment (facucompe) [inline db/migrate/20180920205430_add_devise_token_auth_info_to_users.rb:3]: Be careful in this migration. If you have users in your db you shuold assign them a uid and a provider

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :null_session
+
+  include DeviseTokenAuth::Concerns::SetUserByToken
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
```
Comment (facucompe) [inline app/controllers/application_controller.rb:6]: There is no need to permit this params, the default params are `email` and `password`

**Review by facucompe — CHANGES_REQUESTED** [review]

**Review by facucompe — APPROVED** [review]
Sorry. I didn't realize that the params are for the sign up

**Pull request merged**